src/adapter/etl-adapter-google-sheet/src/Flow/ETL/Adapter/GoogleSheet/GoogleSheetExtractor.php

```diff
@@ -68,14 +68,9 @@ public function extract(FlowContext $context) : \Generator
                 function (array $rowData) use ($headers, $headersCount, $shouldPutInputIntoRows) {
                     $rowDataCount = \count($rowData);
 
-                    if ($headersCount > $rowDataCount) {
-                        \array_push(
-                            $rowData,
-                            ...\array_map(
-                                static fn (int $i) => null,
-                                \range(1, $headersCount - $rowDataCount)
-                            )
-                        );
+                    // Expand columns to the size of the previous row
+                    for ($i = $rowDataCount; $i < $headersCount; $i++) {
```

Comment:
what will happen in situation like this: 

- row 1 - 5 columns
- row 2 - 5 columns
- row 3 - 7 columns
- row 4 - 5 columns

the final will be like below? 

- row 1 - 5 columns
- row 2 - 5 columns
- row 3 - 7 columns
- row 4 - 7 columns

Reply:
I think right now: yes.

This will be similar case like with Google, when we can enforce drop of later extra cells or hardly fail.

Reply:
In that case I would skip that, users can simply use `df()->read()->select(...)` to make sure that only columns they need are present int the output and anything extra is removed, or they can use schema infer where those additional columns are going to be nullable and then they can pass inferred schema to extractor